### PR TITLE
Add android x86_64 user struct

### PR DESCRIPTION
**Pull request by s1341**
This PR adds the `user`, `user_regs_struct` and `user_fpregs_struct` structures to android x86_64

### CODE REVIEW

**Comment by s1341** [comment]
I’m not sure I understand what the failure is. Can you help me?

**Comment by Amanieu** [comment]
You need to remove the existing definition of `PTRACE_GETREGS`/`PTRACE_SETREGS` in `src/unix/linux_like/android/b32/mod.rs` since they are shadowed by this definition.

**Comment by s1341** [comment]
@Amanieu can you try again please?

**Comment by Amanieu** [comment]
It seems that these types use `long` instead of `long long` on Android.

**Comment by s1341** [comment]
@Amanieu I think I fixed it. Can you trigger another build please.

**Comment by Amanieu** [comment]
It seems the registers are also defined as `long long`. Perhaps check against the actual definition in the android headers?

**Comment by s1341** [comment]
Ok. I think I caught everything now, as per [the bionic sources.](https://cs.android.com/android/platform/superproject/+/master:bionic/libc/include/sys/user.h;bpv=1;bpt=1;l=124?q=user_regs_struct&ss=android%2Fplatform%2Fsuperproject&gsn=user_regs_struct&gs=kythe%3A%2F%2Fandroid.googlesource.com%2Fplatform%2Fsuperproject%3Flang%3Dc%252B%252B%3Fpath%3Dbionic%2Flibc%2Finclude%2Fsys%2Fuser.h%23user_regs_struct%2523c%2523e6afNad7HDT&gs=kythe%3A%2F%2Fandroid.googlesource.com%2Fplatform%2Fsuperproject%3Flang%3Dc%252B%252B%3Fpath%3Dbionic%2Flibc%2Finclude%2Fsys%2Fuser.h%23user_regs_struct%2523c%2523o%2524MgfCKPkVq)

**Comment by s1341** [comment]
@Amanieu sorry for all the failed attempts. Can we try this commit please?

**Comment by s1341** [comment]
Thanks! Please let me know when this is in a release...

**Comment by Amanieu** [comment]
If you need these functions in a release, please submit a PR to bump the libc version. I will then publish a new version of the crate.

File: src/unix/linux_like/android/b64/x86_64/mod.rs [hunk 1]
```diff
@@ -254,6 +323,60 @@ cfg_if! {
...
+                // FIXME: .field("xmm_space", &self.xmm_space)
+                // Ignore padding field
+                    .finish()
+            }
+        }
+
+        impl ::hash::Hash for user_fpregs_struct {
+            fn hash<H: ::hash::Hasher>(&self, state: &mut H) {
+                self.cwd.hash(state);
+                self.ftw.hash(state);
```
Comment (Minoru) [inline src/unix/linux_like/android/b64/x86_64/mod.rs:369]: Sorry I'm so late with this, but why is `swd` missing here? It's included in `PartialEq`, so I guess it's a bug, but I'm not sure so I won't submit a PR to change this.

Note also that `swd` is missing from the `Debug` impl.

Comment (Amanieu) [reply src/unix/linux_like/android/b64/x86_64/mod.rs:369]: I think this is an oversight. A PR would be appreciated.

**Pull request merged**